### PR TITLE
test-bot: bugs fix

### PR DESCRIPTION
**Pull request by xu-cheng**
Fixes the bugs revealed in https://github.com/Homebrew/homebrew/pull/36926

In long term, I plan to rewrite test-bot and split it into multiple files. So it's easier to maintain. (There are just  too many monkey patches).

### CODE REVIEW

**Comment by xu-cheng** [comment]
cc @mikemcquaid

File: Library/Homebrew/cmd/test-bot.rb [hunk 1]
```diff
@@ -406,7 +412,8 @@ def formula formula_name
         CompilerSelector.select_for(formula)
       rescue CompilerSelectionError => e
         unless installed_gcc
-          test "brew", "install", "gcc"
+          # Don't care about e.g. bottle failures
+          run_as_not_developer { test "brew", "install", "gcc" }
```
Comment (mikemcquaid) [inline Library/Homebrew/cmd/test-bot.rb]: I'd say the actual fix here is to only do this if the `gcc` formula has been changed?

**Comment by mikemcquaid** [comment]
> In long term, I plan to rewrite test-bot and split it into multiple files. So it's easier to maintain. (There are just too many monkey patches).

I don't think splitting into multiple files makes it easier to maintain, personally. I'm up for cleaning this up further, though. What monkey patches?

File: Library/Homebrew/cmd/test-bot.rb (hunk 1 above)
Comment (xu-cheng) [inline Library/Homebrew/cmd/test-bot.rb]: You are right.

**Comment by xu-cheng** [comment]
I'm just thinking to split different commands into multiple files or methods (like ci-upload and test are irrelevant to each other).

> What monkey patches?

Some duplicated codes can be combined into method. And cleanup some variables.

**Comment by mikemcquaid** [comment]
> Some duplicated codes can be combined into method. And cleanup some variables.

Ok, sorry, I've always understood monkey patching as a Ruby-specific thing where you modify the behaviour of existing classes.

**Comment by xu-cheng** [comment]
Updated.

**Comment by mikemcquaid** [comment]
:+1:

**Pull request closed without merging**